### ISSUE TITLE
with-intercepted-logging intercepts too much with #:logger

### ISSUE THREAD

**Issue by camoy** [racket/racket#3167]
The documentation for `with-intercepted-logging` states:

> If #:logger is specified, intercepts events sent to that logger, otherwise uses a new child logger of the current logger.

But it seems that it captures not only events sent to that logger, but anything sent to it and `(current-logger)`.

```rkt
(define-logger bar)
(with-intercepted-logging
  displayln
  (λ () (log-warning "hello"))
  #:logger bar-logger
  'warning)
```

I thought this wouldn't print anything (since we never invoke `log-bar-warning`), but it does. This is because `with-intercepted-logging` [sets the `current-logger`](https://github.com/racket/racket/blob/master/racket/collects/racket/logging.rkt#L55) to the passed logger. This seems, at least to me, to be the wrong behavior.

**Comment by bennn** [comment]
That's because:

> Every event reported to a logger is propagated to its parent (if any)

https://docs.racket-lang.org/reference/logging.html

**Comment by camoy** [comment]
Yes, but I'm not sure that's at play here. The `bar` logger is a child of the default logger. So my expectation would be that if I intercepted logging events for the default logger I would also get `bar` events. However, the opposite is happening here. I'm only intercepting logging events from `bar`, but I'm also getting events that should've gone to the parent.

I think the core issue is that setting the `current-logger` on [L55](https://github.com/racket/racket/blob/master/racket/collects/racket/logging.rkt#L55) made sense before #1487, but no longer does. Concretely, I ran into this problem when I was trying to intercept debug events from a particular logger, but was getting all of Racket's internal debug events as well. This doesn't seem like it's the intended behavior, but maybe I'm missing something.

**Comment by bennn** [comment]
Ok, I see the confusion. I don't have an opinion yet; I'm not sure where `log-warning` events are supposed to go.

Every time I've used `with-intercepted-logging` has been to test that _at least_ my "bar" messages go out when they should. The `make-log-interceptor` package helps me filter out other events.

**Comment by camoy** [comment]
That's a cool package, I hadn't seen that before! I guess there's two options here:

1. Update the documentation to reflect the fact that `with-intercepted-logging` sets the `current-logger`.
2. Change the implementation to not set the `current-logger` when given `#:logger`. If we don't do this, then we need a way for the log receiver to determine if the messages it gets actually came from the dynamic extent of procedure you give `with-intercepted-logging` (I assume that's why the parameterization is there in the first place). Seems like we could use a continuation mark for that.

Maybe @stamourv has thoughts on this?
